src/tree_search.cpp

```diff
@@ -341,15 +341,15 @@ std::unique_ptr<Node> find_best_split(const std::vector<flat_set>& sorted_sets,
       return level_zero_learning(sorted_sets, data);
     } else {
           // "pruning", the recursive case (same action in both leaves):
-      if ((best_left_child->action_id != -1) &&
+      if ((best_left_child->left_child == nullptr && best_right_child->left_child == nullptr) &&
```

Comment:
And can you make a note of this in the Node struct in tree_search.h? That the convention is to set them to zero...

And what about making lef_child == nullptr a method of Node?

Reply:
I did consider is_leaf and I think that is a great idea. I'll do that.